server/src/main/java/org/elasticsearch/action/admin/indices/alias/get/GetAliasesResponse.java

```diff
@@ -75,5 +126,158 @@ public void writeTo(StreamOutput out) throws IOException {
+        return builder.build();
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+
+        final boolean namesProvided = (params.param("name") != null);
+        final ImmutableOpenMap<String, List<AliasMetaData>> aliasMap = this.aliases;
+
+        final Set<String> indicesToDisplay = new HashSet<>();
+
+        if (namesProvided) {
+            for (final ObjectObjectCursor<String, List<AliasMetaData>> cursor : aliasMap) {
+                if (cursor.value != null && false == cursor.value.isEmpty()) {
+                    indicesToDisplay.add(cursor.key);
+                }
+            }
+        }
+        
+        builder.startObject();
+        {
+            if (status != null && RestStatus.OK != status) {
+                builder.field("error", errorMsg);
+                builder.field("status", status.getStatus());
+            }
+
+            for (final ObjectObjectCursor<String, List<AliasMetaData>> entry : aliases) {
+                if (namesProvided == false || indicesToDisplay.contains(entry.key)) {
+                    builder.startObject(entry.key);
+                    {
+                        builder.startObject("aliases");
+                        {
+                            for (final AliasMetaData alias : entry.value) {
+                                AliasMetaData.Builder.toXContent(alias, builder, ToXContent.EMPTY_PARAMS);
+                            }
+                        }
+                        builder.endObject();
+                    }
+                    builder.endObject();
+                }
+            }
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    public static GetAliasesResponse fromXContent(XContentParser parser) throws IOException {
+        if (parser.currentToken() == null) {
+            parser.nextToken();
+        }
+        ensureExpectedToken(Token.START_OBJECT, parser.currentToken(), parser::getTokenLocation);
+        ImmutableOpenMap.Builder<String, List<AliasMetaData>> aliasesBuilder = ImmutableOpenMap.builder();
+
+        String currentFieldName;
+        Token token;
+        String exceptionMsg = null;
+        RestStatus status = RestStatus.OK;
+        ElasticsearchException exception = null;
+
+        while ((token = parser.nextToken()) != Token.END_OBJECT) {
+            if (parser.currentToken() == Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+
+                if ("status".equals(currentFieldName)) {
+                    if ((token = parser.nextToken()) != Token.FIELD_NAME) {
+                        ensureExpectedToken(XContentParser.Token.VALUE_NUMBER, token, parser::getTokenLocation);
+                        status = RestStatus.fromCode(parser.intValue());
+                    }
+                } else if ("error".equals(currentFieldName)) {
+                    if ((token = parser.nextToken()) != Token.FIELD_NAME) {
```

Comment:
this resembles what `ElasticsearchException#failureFromXContent` does. but it will build an exception in the first case with a string message only. That's why I was hoping that we could use an exception in every case, but I don't think it would work at transport, as such exception would be serialized then completely different when calling toXContent on it. There's `generateFailureXContent` that is similar to what we would need, yet not quite what we need. I think what you have is good, unless you want to reuse the parsing code that we already have and then take the string out of the parsed exception which is what you need. Not sure really.

Reply:
Thanks for the tip! I will have a look and if I can reuse the `failureFromXContent` or `generateFailureXContent`.